src/testnet/burnchain.rs

```diff
@@ -0,0 +1,141 @@
+use burnchains::bitcoin::BitcoinBlock;
+use chainstate::burn::db::burndb::{BurnDB};
+use chainstate::burn::{BlockSnapshot};
+use chainstate::burn::operations::{BlockstackOperationType};
+use util::hash::Sha256Sum;
+
+/// BurnchainSimulator is simulating a simplistic burnchain.
+/// 
+/// When calling start(), a thread is being detached, looping, sleeping and firing some events, and a tuple (tx, rx) 
+/// is being returned so that the caller can be receiving new blocks, and transmitting ops in a simulated mem_pool.
+pub struct BurnchainSimulator {
+    mem_pool: Arc<Mutex<Vec<BlockstackOperationType>>>,
+    db: Option<Arc<Mutex<BurnDB>>>,
+}
+
+impl BurnchainSimulator {
+
+    pub fn new() -> Self {
+        Self {
+            mem_pool: Arc::new(Mutex::new(vec![])),
+            db: None,
+        }
+    }
+    
+    pub fn start(&mut self, config: &Config) -> (mpsc::Receiver<(BlockSnapshot, Vec<BlockstackOperationType>, Arc<Mutex<BurnDB>>)>, mpsc::Sender<BlockstackOperationType>) {
+        let (block_tx, block_rx) = mpsc::channel();
+        
+        let path = config.burnchain_path.clone();
+        let chain = config.chain.clone();
+        let name = config.testnet_name.clone();
+        let block_time = time::Duration::from_millis(config.burnchain_block_time);
+
+        let ops_dequeuing = Arc::clone(&self.mem_pool);
+        let mut vtxindex = 1;
+
+        let db = BurnDB::connect(&path, 0, &BurnchainHeaderHash([0u8; 32]), true).unwrap();
+        self.db = Some(Arc::new(Mutex::new(db)));
+        let burn_db = Arc::clone(&self.db.as_ref().unwrap());
+
+        thread::spawn(move || {
+
+            let chain = Burnchain::new(&path, &chain, &name).unwrap();
+            
+            let mut chain_tip = {
+                let db = burn_db.lock().unwrap();
+                BurnDB::get_first_block_snapshot(db.conn()).unwrap()
+            };
+
+            block_tx.send((chain_tip.clone(), vec![], Arc::clone(&burn_db))).unwrap();    
+                
+            loop {
+                thread::sleep(block_time);
+
+                // Simulating mining
+                let next_block_header = BurnchainSimulator::build_next_block_header(&chain_tip);
+
+                // Updating ops properties before including them in the new block
+                let mut ops_to_include = vec![];
+                {
+                    let mut ops = ops_dequeuing.lock().unwrap();
+                    for op in ops.iter_mut() {
+                        match op {
+                            BlockstackOperationType::LeaderKeyRegister(ref mut op) => {
+                                op.block_height = next_block_header.block_height;
+                                op.burn_header_hash = next_block_header.block_hash;
+                                op.vtxindex = vtxindex;
+                                op.txid = Txid(Sha256Sum::from_data(format!("{}", vtxindex).as_bytes()).0);
```

Comment:
To make the simulation more accurate, `vtxindex` should be reset to `1` for each block, since it represents the index into the transaction array where the Blockstack transaction was found.  In practice, two transactions can have the same vtxindex.

To keep ensuring that `op.txid` is globally unique, you could mix the `vtxindex` and `block_height` together, since `(block_height, vtxindex)` is unique to the fork.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1202/commits/67020104b6bdd172d980e1040c96d20db7d9687a